File: Files/Docs/Мультиметры управление etc/ADL-GPIB/Samples/BCB/Change GPIB Address/Unit1.cpp
/********************************************************************
*   Example: Change GPIB Address
*
*   Description: Get/set the address of GPIB interface card using ibask/ibpad/ibsad commands.
*
*   ADLINK Inc. 2006
********************************************************************/
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Unit1.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------


void __fastcall TForm1::btnGetClick(TObject *Sender)
{
  //declare variables
  int rval;
  AnsiString str;

  //Read the primary GPIB address of GPIB interface card using ibask command
  ibask (ud, IbaPAD, &rval);
  if (ibsta & ERR)
  {
    ShowMessage("Error in querying the primary address of GPIB interface card.");
  }

  str.sprintf("%d", rval);
  EditPadGet->Text=str;

  //Read the secondary GPIB address of GPIB interface card using ibask command
  ibask (ud, IbaSAD, &rval);
  if (ibsta & ERR)
  {
    ShowMessage("Error in querying the secondary address of GPIB interface card.");
  }
  str.sprintf("%d", rval);
  EditSadGet->Text=str;

}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnSetClick(TObject *Sender)
{

  //Set the primary GPIB address of GPIB interface card using ibpad command
  ibpad(ud, atoi(EditPadSet->Text.c_str()));
  if (ibsta & ERR)
  {
    ShowMessage("Error in setting the primary address of GPIB interface card.");
  }

  //Set the secondary GPIB address of GPIB interface card using ibsad command
  ibsad(ud, atoi(EditSadSet->Text.c_str()));
  if (ibsta & ERR)
  {
    ShowMessage("Error in setting the secondary address of GPIB interface card.");
  }

}
//---------------------------------------------------------------------------
void __fastcall TForm1::cmbGPIBClick(TObject *Sender)
{
  //declare variables
  AnsiString str;

//    Offline the GPIB interface card
  ibonl( ud, 0);

  str.sprintf("GPIB%d", cmbGPIB->ItemIndex);
  //Find the GPIB interface card
  ud = ibfind(str.c_str());
  if (ibsta & ERR)
  {
    ShowMessage("Error in finding the GPIB interface card.");
  }

}
//---------------------------------------------------------------------------

File: Files/SRC/agilent 3458a/Change GPIB Address/Unit1.pas
(********************************************************************
*   Example:Change GPIB Address
*
*   Description: Get/set the address of GPIB interface card using ibask/ibpad/ibsad commands.
*
*   ADLINK Inc. 2006
********************************************************************)	
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls, gpib_user, Adgpib;

type
  TForm1 = class(TForm)
    Label1: TLabel;
    cmbGPIB: TComboBox;
    GroupBox1: TGroupBox;
    GroupBox2: TGroupBox;
    editPadGet: TEdit;
    Label2: TLabel;
    Label3: TLabel;
    Label4: TLabel;
    Label5: TLabel;
    editPadSet: TEdit;
    editSadGet: TEdit;
    editSadSet: TEdit;
    btnGet: TButton;
    btnSet: TButton;
    Label6: TLabel;
    Label7: TLabel;
    procedure FormCreate(Sender: TObject);
    procedure btnGetClick(Sender: TObject);
    procedure btnSetClick(Sender: TObject);
    procedure cmbGPIBClick(Sender: TObject);
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

  (*488.2 GPIB global status variables.*)
  ibsta  : Integer;
  iberr  : Integer;
  ibcnt  : Integer;
  ibcntl : Integer;
  
  ud: Integer;

implementation

{$R *.DFM}

procedure TForm1.FormCreate(Sender: TObject);
var
  str: AnsiString ;

begin
  cmbGPIB.ItemIndex := 0;
  str := Format('GPIB%d', [cmbGPIB.ItemIndex]);
  //Find the GPIB interface card
  ud := ibfind(PChar(str));
  gpib_get_globals(@ibsta, @iberr, @ibcnt, @ibcntl);
  if (ibsta and ERR <> 0) Then
  begin
    MessageDlg('Error in finding the GPIB interface card.', mtError, [mbOK], 0);
  end;

end;

procedure TForm1.btnGetClick(Sender: TObject);
var
  //declare variables
  rval: integer ;
  str: AnsiString ;

begin

  //Read the primary GPIB address of GPIB interface card using ibask command
  ibask (ud, IbaPAD, @rval);
  gpib_get_globals(@ibsta, @iberr, @ibcnt, @ibcntl);
  if (ibsta and ERR <> 0) Then
  begin
    MessageDlg('Error in querying the primary address of GPIB interface card.', mtError, [mbOK], 0);
  end;

  str := Format('%d', [rval]);
  EditPadGet.Text := str;

  //Read the secondary GPIB address of GPIB interface card using ibask command
  ibask (ud, IbaSAD, @rval);
  gpib_get_globals(@ibsta, @iberr, @ibcnt, @ibcntl);
  if (ibsta and ERR <> 0) then
  Begin
    MessageDlg('Error in querying the secondary address of GPIB interface card.', mtError, [mbOK], 0);
  end;
  str := format('%d', [rval]);
  EditSadGet.Text := str;

end;

procedure TForm1.btnSetClick(Sender: TObject);
var
  //declare variables
  ud, rval: integer ;
  str: AnsiString ;
begin
  str := Format('GPIB%d', [cmbGPIB.ItemIndex]);
  //Find the GPIB interface card
  ud := ibfind(PChar(str));

  //Set the primary GPIB address of GPIB interface card using ibpad command
  ibPad(ud, StrToInt(EditpadSet.Text));
  gpib_get_globals(@ibsta, @iberr, @ibcnt, @ibcntl);
  if (ibsta and ERR <> 0) Then
  begin
    MessageDlg('Error in setting the primary address of GPIB interface card.', mtError, [mbOK], 0);
  end;


  //Set the secondary GPIB address of GPIB interface card using ibsad command
  ibsad(ud, StrToInt(EditSadSet.Text));
  gpib_get_globals(@ibsta, @iberr, @ibcnt, @ibcntl);
  if (ibsta and ERR <> 0) then
  Begin
    MessageDlg('Error in setting the secondary address of GPIB interface card.', mtError, [mbOK], 0);
  end;


end;

procedure TForm1.cmbGPIBClick(Sender: TObject);
var
  str: AnsiString ;

begin

  //    Offline the GPIB interface card
  ibonl( ud, 0);

  str := Format('GPIB%d', [cmbGPIB.ItemIndex]);
  //Find the GPIB interface card
  ud := ibfind(PChar(str));
  gpib_get_globals(@ibsta, @iberr, @ibcnt, @ibcntl);
  if (ibsta and ERR <> 0) Then
  begin
    MessageDlg('Error in finding the GPIB interface card.', mtError, [mbOK], 0);
  end;

end;

procedure TForm1.FormClose(Sender: TObject; var Action: TCloseAction);
begin
  //    Offline the GPIB interface card
  ibonl( ud, 0);

end;

end.
